### DISCUSSION TITLE
Get polynomial equation

### DISCUSSION THREAD

**Discussion by D0nVitalio** [PHPOffice/PhpSpreadsheet#2342] [Q&A]
Hi. I need code which accept array of coordinates [[x1,y1],[xn+2,yn+1],..] and returns polynomial equation.
Found this:
http://polynomialregression.drque.net/  which gives me f( x ) = 6 x3 - 6.25 x2 + 0.6 x + 0.65

Does https://github.com/PHPOffice/PhpSpreadsheet/blob/master/src/PhpSpreadsheet/Shared/Trend/PolynomialBestFit.php
can take coordinates and return polynomial equation?

**Comment by oleibman** [comment]
Sorry for the much-delayed, and probably disappointing, answer. PolynomialBestFit in theory is supposed to provide what you need. However, it doesn't work, and, if you try to use it, it will just throw an exception.

The solution to which you point is interesting. I will study it further, but it has at least two problems for us. One is that it is not in Composer, so we would have to maintain it on our own (including adding our own tests), and I'm not convinced that that is worthwhile. After all, you could just add it to your own project and  apply its results to PhpSpreadsheet however you wish. The other problem is that it requires BCMath, which PhpSpreadsheet does not. We could add that as a requirement or an optional requirement, but I'm again not convinced that's worthwhile.
